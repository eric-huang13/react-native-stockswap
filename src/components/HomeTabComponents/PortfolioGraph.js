import React, {Component} from 'react';
import {StyleSheet, Dimensions, SafeAreaView} from 'react-native';
import {SlideAreaChart} from '@connectedcars/react-native-slide-charts';
import {LinearGradient, Stop} from 'react-native-svg';

export default class PortfolioGraph extends Component {
  render() {
    const lineColor = this.props.percent > 0 ? '#91f2b1' : '#F66E6E';
    const fillGradient = (props) => {
      return (
        <LinearGradient x1="40%" y1="0%" x2="40%" y2="100%" {...props}>
          <Stop stopColor={lineColor} offset="5%" stopOpacity="0.1" />
          <Stop stopColor="#4a334a" offset="100%" stopOpacity="0" />
        </LinearGradient>
      );
    };

    return (
      <SafeAreaView style={styles.container}>
        <SlideAreaChart
          data={this.props.graphData}
          yRange={this.props.range}
          width={Dimensions.get('window').width - 180}
          height={70}
          style={{backgroundColor: '#2a334a'}}
          axisWidth={5}
          axisHeight={5}
          chartLineColor={lineColor}
          throttleAndroid={true}
          renderFillGradient={fillGradient}
          alwaysShowIndicator={false}
          cursorProps={{
            displayCursor: false,
          }}
          paddingBottom={10}
          paddingLeft={3}
          //Y AXIS
          yAxisProps={{
            showBaseLine: false,
            verticalLineWidth: 0,
            horizontalLineWidth: 0,
            axisLabelAlignment: 'top',
            rotateAxisLabel: true,
            labelTopPadding: 0,

            //Set graph Y numbers
            numberOfTicks: 0,
          }}
        />
      </SafeAreaView>
    );
  }
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
