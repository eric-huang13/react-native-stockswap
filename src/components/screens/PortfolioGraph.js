import React, {Component} from 'react';
import {StyleSheet, Dimensions, SafeAreaView} from 'react-native';
import {SlideAreaChart} from 'react-native-slide-charts';
import {LinearGradient, Stop} from 'react-native-svg';

export default class PortfolioGraph extends Component {
  render() {
    const fillGradient = (props) => {
      return (
        <LinearGradient x1="40%" y1="0%" x2="40%" y2="100%" {...props}>
          <Stop stopColor={'#91f2b1'} offset="5%" stopOpacity="0.1" />
          <Stop stopColor="#4a334a" offset="100%" stopOpacity="0" />
        </LinearGradient>
      );
    };

    return (
      <SafeAreaView style={styles.container}>
        <SlideAreaChart
          data={this.props.graphData}
          yRange={this.props.range}
          width={Dimensions.get('window').width - 162}
          height={100}
          style={{backgroundColor: '#2a334a'}}
          axisWidth={5}
          axisHeight={5}
          chartLineColor={'#91f2b1'}
     
          throttleAndroid={true}
          renderFillGradient={fillGradient}
          alwaysShowIndicator={false}
          cursorProps={{
            displayCursor: false,
           
          }}
          paddingBottom={10}
          paddingLeft={3}
          //Y AXIS
          yAxisProps={{
            showBaseLine: false,
            verticalLineWidth: 0,
            horizontalLineWidth: 0,
            axisLabelAlignment: 'top',
            rotateAxisLabel: true,
            labelTopPadding: 0,

            //Set graph Y numbers
            numberOfTicks: 0,
           
          }}

        />
      </SafeAreaView>
    );
  }
}
const styles = StyleSheet.create({
  container: {
      flex: 1,
  
  },
});
