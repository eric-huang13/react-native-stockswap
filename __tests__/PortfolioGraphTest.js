import React from 'react';
import PortfolioGraph from '../src/components/HomeTabComponents/PortfolioGraph';
import ShallowRenderer from 'react-test-renderer/shallow'; // ES6


jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
jest.mock('react-redux', () => ({
    connect: () => {
      return (component) => {
        return component
      };
    }
  }));
  

test('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render   (<PortfolioGraph />);
    const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});