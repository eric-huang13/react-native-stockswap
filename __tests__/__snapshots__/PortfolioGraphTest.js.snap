// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "flex": 1,
    }
  }
>
  <SlideAreaChart
    alwaysShowIndicator={false}
    animated={true}
    axisHeight={5}
    axisWidth={5}
    chartLineColor="#F66E6E"
    chartLineWidth={3}
    chartPaddingTop={16}
    cursorProps={
      Object {
        "displayCursor": false,
      }
    }
    curveType={[Function]}
    data={Array []}
    height={70}
    paddingBottom={10}
    paddingLeft={3}
    paddingRight={8}
    paddingTop={8}
    renderFillGradient={[Function]}
    shouldCancelWhenOutside={true}
    style={
      Object {
        "backgroundColor": "#2a334a",
      }
    }
    throttleAndroid={true}
    width={570}
    xScale="linear"
    yAxisProps={
      Object {
        "axisLabelAlignment": "top",
        "horizontalLineWidth": 0,
        "labelTopPadding": 0,
        "numberOfTicks": 0,
        "rotateAxisLabel": true,
        "showBaseLine": false,
        "verticalLineWidth": 0,
      }
    }
  />
</ForwardRef(SafeAreaView)>
`;
